### PR TITLE
Add ApiOption overloads to IIssuesClient and IObservableIssuesClient

### LINKED ISSUE #1158: Add ApiOption overloads to methods on IIssuesClient
As part of completing #760 a number of overloads need to be added to the client interfaces.

As an example, this has been added to [`IReleasesClient`](https://github.com/octokit/octokit.net/commit/03ea83148995711fc47148489e2123e65f33bd82)and [`IObservableReleasesClient`](https://github.com/octokit/octokit.net/commit/aa3ce0db9210b778414e290af7f17c4a646bdeec).

The checklist for porting each method is:

- Add an overload for this method on `IIssuesClient` with a new parameter `ApiOptions`
- Implement the new method on the class which implements `IIssuesClient`
- Add an overload for this method on `IObservableIssuesClient` with a new parameter `ApiOptions`
- Implement the new method on the class which implements `IObservableIssuesClient`

This needs to be done for each of these methods:

- [ ] `Task<IReadOnlyList<Issue>> GetAllForCurrent()`
- [ ] `Task<IReadOnlyList<Issue>> GetAllForCurrent(IssueRequest request)`
- [ ] `Task<IReadOnlyList<Issue>> GetAllForOrganization(String organization)`
- [ ] `Task<IReadOnlyList<Issue>> GetAllForOrganization(String organization, IssueRequest request)`
- [ ] `Task<IReadOnlyList<Issue>> GetAllForOwnedAndMemberRepositories()`
- [ ] `Task<IReadOnlyList<Issue>> GetAllForOwnedAndMemberRepositories(IssueRequest request)`
- [ ] `Task<IReadOnlyList<Issue>> GetAllForRepository(String owner, String name)`
- [ ] `Task<IReadOnlyList<Issue>> GetAllForRepository(String owner, String name, RepositoryIssueRequest request)`

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #1158

This is a big one, so I'm going through it as slowly and methodically as possible:

 - [x] add new methods
 - [x] update impacted unit tests 
 - [x] add new unit tests wherever appropriate
 - [x] add new integration tests wherever appropriate
 - [x] correct references in xml-docs

### CODE REVIEW

File: Octokit.Tests/Clients/IssuesClientTests.cs [hunk 1]
```diff
@@ -60,13 +71,29 @@ public void SendsAppropriateParameters()
...
+                var client = new IssuesClient(Substitute.For<IApiConnection>());
+
+                await Assert.ThrowsAsync<ArgumentNullException>(
+                    () => client.GetAllForOwnedAndMemberRepositories((ApiOptions)null));
+                await Assert.ThrowsAsync<ArgumentNullException>(
+                    () => client.GetAllForOwnedAndMemberRepositories((IssueRequest)null));
+                await Assert.ThrowsAsync<ArgumentNullException>(
+                    () => client.GetAllForOwnedAndMemberRepositories(null, new ApiOptions()));
+                await Assert.ThrowsAsync<ArgumentNullException>(
+                    () => client.GetAllForOwnedAndMemberRepositories(new IssueRequest(), null));
```
Comment (ryangribble) [inline Octokit.Tests/Clients/IssuesClientTests.cs]: really minor nitpick but the equivalent code in other unit tests in this file are all on one line, while these ones are indented/split...

File: Octokit.Tests/Clients/IssuesClientTests.cs [hunk 2]
```diff
@@ -99,30 +151,87 @@ public void SendsAppropriateParameters()
...
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllForRepository("owner", null, request, options));
+
+                await Assert.ThrowsAsync<ArgumentException>(() => client.GetAllForRepository("owner", ""));
+                await Assert.ThrowsAsync<ArgumentException>(() => client.GetAllForRepository("owner", "", options));
+                await Assert.ThrowsAsync<ArgumentException>(() => client.GetAllForRepository("owner", "", request));
+                await Assert.ThrowsAsync<ArgumentException>(() => client.GetAllForRepository("owner", "", request, options));
+
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllForRepository("owner", "name", (ApiOptions)null));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllForRepository("owner", "name", (RepositoryIssueRequest)null));
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllForRepository("owner", "name", null, options));
```
Comment (ryangribble) [inline Octokit.Tests/Clients/IssuesClientTests.cs]: are we missing one case here where the owner, name and request are provided, but `null` for the options?

File: Octokit.Tests/Clients/IssuesClientTests.cs (hunk 1 above)
Comment (shiftkey) [inline Octokit.Tests/Clients/IssuesClientTests.cs]: :thumbsup: just my 80-chars-per-line OCD acting up. I'll undo this.

File: Octokit.Tests/Clients/IssuesClientTests.cs (hunk 2 above)
Comment (shiftkey) [inline Octokit.Tests/Clients/IssuesClientTests.cs]: yes, yes we are

File: Octokit.Tests/Reactive/ObservableIssuesClientTests.cs [hunk 3]
```diff
@@ -36,9 +36,42 @@ public void GetsFromClientIssueIssue()
...
+            await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllForRepository("owner", null, request, options).ToTask());
+
+            await Assert.ThrowsAsync<ArgumentException>(() => client.GetAllForRepository("owner", "").ToTask());
+            await Assert.ThrowsAsync<ArgumentException>(() => client.GetAllForRepository("owner", "", options).ToTask());
+            await Assert.ThrowsAsync<ArgumentException>(() => client.GetAllForRepository("owner", "", request).ToTask());
+            await Assert.ThrowsAsync<ArgumentException>(() => client.GetAllForRepository("owner", "", request, options).ToTask());
+
+            await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllForRepository("owner", "name", (ApiOptions)null).ToTask());
+            await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllForRepository("owner", "name", (RepositoryIssueRequest)null).ToTask());
+            await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllForRepository("owner", "name", null, options).ToTask());
```
Comment (ryangribble) [inline Octokit.Tests/Reactive/ObservableIssuesClientTests.cs]: here too (case where `options` is `null`)

**Comment by ryangribble** [comment]
it's hard to review these changes, but Im grasping at straws to find anything major here 😀

**Comment by shiftkey** [comment]
@ryangribble thanks for the :eyes:, I think there's a bunch of new test coverage here I can do but given how many overloads there are I'm trying to keep this as focused as possible.

**Comment by shiftkey** [comment]
I also took some time out in 1b2d009e3a33b966fa78e0b3b8883a7863797bd6 to document the return values from the API. Let me know if you have any feedback on the words.

**Comment by shiftkey** [comment]
Lots of the pagination endpoints here require you to have org access, so rather than getting too bogged down in pagination I'm going to skip those tests.

**Comment by ryangribble** [comment]
> I also took some time out in 1b2d009 to document the return values from the API. Let me know if you have any feedback on the words.

I noticed you are using `<seealso cref=""` when most other cases in the code base are using `<see cref=""` - is there any reason for using one over the other?

The wording sounds reasonable although I wonder whether the effort involved to doc all return values is worth it, since they are pretty self evident... and it then sorta means we have to go and add them to allll the rest of the codebase.  I guess the same could be said for doccing most parameters in general though and if we didnt do it then we would have no doc anywhere.  Does this mean you intend us to start the process of docing return value on everything from now on?

**Comment by shiftkey** [comment]
Can I get a :thumbsup: here before I close this out?

**Pull request merged**